### ISSUE TITLE
Question of localization in functions.php

### ISSUE THREAD

**Issue by stefaweb** [mailwatch/1.2.0#450]
Don't know if I can safety localize this in functions.php?

return 'spam';
return 'not spam';

and

autolearn=spam
autolearn=not spam

```
/**
 * @param $spamreport
 * @return string|false
 */
function sa_autolearn($spamreport)
{
    switch (true) {
        case(preg_match('/autolearn=spam/', $spamreport)):
            return 'spam';
        case(preg_match('/autolearn=not spam/', $spamreport)):
            return 'not spam';
        default:
            return false;
    }
}
```

![capture d ecran 2017-01-28 a 11 38 33](https://cloud.githubusercontent.com/assets/12907102/22396021/84b90d0c-e54e-11e6-9427-d61bfdb4ccf1.png)

![capture d ecran 2017-01-28 a 11 38 55](https://cloud.githubusercontent.com/assets/12907102/22396024/8a09bda6-e54e-11e6-9155-18ddaa66cb2f.png)

**Comment by asuweb** [comment]
Yes, you could localise the return values.   it - it's only used in detail.php - line 219 - 225.

**Comment by Skywalker-11** [comment]
as a tip: if you want to find where a function is used use `grep -r <functionname> <mailmatchdir>`

**Comment by stefaweb** [comment]
ok,

One of you can add this in one of your PR where you edit lang files?

I don't want to open a PR just for that.

**Comment by asuweb** [comment]
I'll do it the one I've just opened - It changes only the relevant files to this pr

**Comment by asuweb** [comment]
Fixed

**Comment by stefaweb** [comment]
For fr.php:

```
'saspam03' => 'spam',
'sanotspam03' => 'N\'est pas un spam',
```

**Comment by Skywalker-11** [comment]
de.php:```

'saspam03' => 'Spam',
'sanotspam03' => 'Kein Spam',
```

**Comment by stefaweb** [comment]
Found another one in detail.php

`'&nbsp;&nbsp;Action(s): '`

`'&nbsp;&nbsp;' . __('actionspam04') . ' '`

`'actionspam04' => 'Action(s):',`


```
        if ($fieldn === 'Spam:' && !DISTRIBUTED_SETUP) {
            // Display actions if spam/not-spam
            if ($row[$f] === $yes) {
                $row[$f] = $row[$f] . '&nbsp;&nbsp;Action(s): ' . str_replace(' ', ', ', get_conf_var('SpamActions'));
            } else {
                $row[$f] = $row[$f] . '&nbsp;&nbsp;Action(s): ' . str_replace(
                        ' ',
                        ', ',
                        get_conf_var('NonSpamActions')
                    );
            }
        }
```

**Comment by asuweb** [comment]
Good spot @stefaweb - the `if ($fieldn === 'Spam:'` only works just now by accident it seems.  

No one has changed the localisation for `'spam04'` fortunately, which is why that latent bug hasn't been discovered.

I'll change it. 

I suppose we can localise Actions(s) as well

**Comment by stefaweb** [comment]
Yes, me !

`    'spam04' => 'Spam :',`

colon...

**Comment by asuweb** [comment]
It's fixed in the next commit

**Comment by stefaweb** [comment]
Encore une exception française !

**Comment by asuweb** [comment]
This was merged in #451 - can be closed :)

**Comment by stefaweb** [comment]
Done.
